### PR TITLE
Add stduuid 1.0

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **stduuid/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/stduuid/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    def package(self):
+        root_dir = self._source_subfolder
+        include_dir = os.path.join(root_dir, "include")
+        self.copy(pattern="LICENSE", dst="licenses", src=root_dir)
+        self.copy(pattern="uuid.h", dst="include", src=include_dir)
+
+    def package_info(self):
+        self.cpp_info.defines.append('ASIO_STANDALONE')
+        if tools.os_info.is_linux:
+            self.cpp_info.libs.append('pthread')
```
Comment (Croydon) [inline recipes/stduuid/all/conanfile.py]:
```suggestion
            self.cpp_info.system_libs.append('pthread')
```

File: recipes/stduuid/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+           not any([str(self.settings.compiler.cppstd) == std for std in ["17", "20", "gnu17", "gnu20"]]):
+            raise ConanInvalidConfiguration("stduuid requires at least c++17")
+        elif compiler == "Visual Studio" and \
+            not any([self.settings.compiler.cppstd == std for std in ["17", "20"]]):
+                raise ConanInvalidConfiguration("stduuid requires at least c++17")
+        else:
+            if ( compiler == "gcc" and version < "7" ) or ( compiler == "clang" and version < "5" ):
+                raise ConanInvalidConfiguration("stduuid requires a compiler that supports at least C++17")
+            elif compiler == "apple-clang":
+                self.output.warn("stduuid is not tested with apple-clang")
```
Comment (Croydon) [inline recipes/stduuid/all/conanfile.py]: Stuuid got likely tested with Apple Clang?
I guess this is a line you didn't need to copy over 😃

File: recipes/stduuid/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
```
Comment (uilianries) [inline recipes/stduuid/all/conanfile.py]:
```suggestion
from conans.errors import ConanInvalidConfiguration
from conans.tools import Version
```

**Comment by danimtb** [comment]
@intelligide please address the changes requested

File: recipes/stduuid/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class StduuidConan(ConanFile):
+    name = "stduuid"
+    description = "A C++17 cross-platform implementation for UUIDs"
+    topics = ("conan", "uuid", "guid")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mariusbancila/stduuid"
+    license = "MIT"
+
```
Comment (uilianries) [inline recipes/stduuid/all/conanfile.py]:
```suggestion
    settings = "os", "compiler"
```

**Comment by floriansimon1** [comment]
@intelligide Any idea why Visual Studio < 16 is unsupported? I've had success building my version locally with Visual Studio 15, but I'm guessing it will backfire soon :)

**Comment by intelligide** [comment]
A fix is on the way (#4335) ! Thanks for reporting.

**Comment by floriansimon1** [comment]
Oh wow, I'm impressed. Thanks a lot! I guess I won't need to fork the recipe after all!

**Pull request merged**